### DISCUSSION TITLE
Composite primary (UUID) keys?

### DISCUSSION THREAD

**Discussion by iamjrock** [pgEdge/cli#51] [Q&A] [answered]
pgEdge is awesome, thx team.

For Active-Active replication, I understand that I need a primary key on each table and that I cannot use sequences/auto-increment.  So an UUID, ULID, etc.

But what about composite primary (UUID) keys?

Let's take the example of a many-to-many pivot table:

1. Will pgEdge work with this:
```
[category_id (UUID) | product_id (UUID)] ....... with PrimaryKey[category_id, product_id]
```

3. Or must I do this:
```
[id (UUID) | category_id (UUID) | product_id (UUID)] ....... with PrimaryKey[id]
```

**Comment by iamjrock** [comment]
Any thoughts on this are greatly appreciated please.

**Comment by luss** [comment, marked as answer]
yes you can have composite primary keys.  Also, stay tuned for 64-bit
snowflake sequences that are guaranteed to be unique on your cluster.
Snowflakes are faster to generate the UUID's and they only take upo
64bits rather than 128bits.

**Reply by iamjrock** [reply]
OK great - looking forward to Snowflake sequences!
